### ISSUE TITLE
Cannot override installer hash as admin. Even if EnableLua = 0 (UAC OFF)

### ISSUE THREAD

**Issue by MarcoPeraza** [microsoft/winget-cli#715]
# Cannot override installer hash when running as admin, even if EnableLua = 0 (UAC OFF)

I want to install the package. Please let me install the package.

# Steps to reproduce

Try to install a package with a mismatched hash while running as administrator.

# Expected behavior

It allows you to install.

# Actual behavior

It does not allow you to install.

**Comment by jedieaston** [comment]
Did you run it with --force?

**Comment by denelon** [comment]
@MarcoPeraza were you able to use --force?

**Comment by MarcoPeraza** [comment]
Yes, when I use --force I get the following error still: "Installer hash does not match; this cannot be overridden when running as admin"

Note that when using an admin account and running Windows with EnableLua=0, all applications run with full privileges. This turns UAC off. The hash override should remain available to users running with this configuration. I have never encountered a feature in any Microsoft product that did not work with this configuration.

**Comment by MarcoPeraza** [comment]
Hi, is there any reason why this feature is disabled at all for users running as admin? This seems to be upside-down. Usually, running as administrator gives you _more_ privileges, not _less_.

I also just want to reiterate that when you turn UAC off on a machine, it's impossible to _not_ run as admin, so the hash override feature becomes completely unavailable.

**Comment by jedieaston** [comment]
If you are running as administrator and the installer on the other end is compromised, it can do whatever it wants to your system once winget runs it. If you aren't running as administrator and do --force, UAC should pop up and say "hey, this application wasn't signed by a vendor. run anyway?" 

It should be configurable via ` winget settings `, but by default winget shouldn't be allowing people to easily run unverified binaries (since if there is a hash mismatch, it means that the version that winget downloaded wasn't verified through the community repo and therefore could be a virus or something). 

> I also just want to reiterate that when you turn UAC off on a machine, it's impossible to not run as admin, so the hash override feature becomes completely unavailable.

This is good default behavior. The only reason hash overrides are really needed right now are because the manifest update time is so long, but once the bots get better and we have other repos to choose from this shouldn't be much of an issue.

**Comment by denelon** [comment]
We're looking at several options to deal with this scenario. There have been several concerns about having command-line overrides that could potentially result in malicious code getting installed by the Windows Package Manager (eg. copy-paste this command from some "bad actor"). 

A few options have been suggested:

Have a setting requirement to enable the capability (enables a Group Policy to block behavior by IT administrators).
Change the value to something other than --force (something more akin to --IgnoreSecuritysha256).

Generally, it is preferred not to ignore security checks. Are there any other suggestions here?

**Comment by MarcoPeraza** [comment]
I don't think you need to have that concern. There are any number of things that if you copy and paste into a administrator command prompt will absolutely trash/compromise/etc. your computer. An admin command prompt can do just about anything on a Windows machine (even perform an escalation to SYSTEM). So I don't see a point in being concerned about it in this one small case. Especially if someone has turned off UAC.

**Comment by denelon** [comment]
We now have a Group Policy to determine if the ability to override hash validation is available to users. We are discussing adding a setting to the "settings.json" file to "enable the ability to --force". Once that is in place we can allow either a user or an administrator to override the hash validation.

I'm converting this bug into a feature to capture the work.

**Comment by Zamiell** [comment]
Related: #714 

Question:
I'm trying to automatically install Google Chrome in a post-install script.
The script is running as the SYSTEM user, since it is being invoked by the operating system on the first boot.
When I execute the following command:

```powershell
winget install --accept-source-agreements --silent --exact --id Google.Chrome --force
```

It fails with the following error message:

```
Installer hash does not match; this cannot be overridden when running as admin
```
(As previously mentioned in this issue.)

As a workaround, I tried performing the following:

```powershell
runas /user james-pc\james "winget install --accept-source-agreements --silent --exact --id Google.Chrome --force"
```

This worked, but it requires that:

1) The end-user manually types in the password of the user that they are using with `runas`.
2) A UAC notification pops up that must be manually clicked on.

Thus, this is not a viable workaround, since the point of this script is that it will automatically install Google Chrome without any user intervention.

Does anyone else know of any other workarounds to this problem?

In conclusion, I definitely have to agree with MarcoPeraza - "automatically installing Google Chrome" is probably one of the most basic things that one would expect a competent package manager to do. So the fact that `winget` may not be able to perform this right now is a glaring hole.

**Comment by jedieaston** [comment]
Any other way to work around it would probably be a bug in UAC. 

The actual solution to your problem is that we need to be better about updating Chrome and Spotify (the two apps that change their installer hashes most often) in the community repo. I just saw the PR for it had been sitting for 24 hours or more before a moderator got it approved (oops), then it sat for another hour until the bot got it merged. I've gotten lazy due to the community automation being better than it used to be, but it still misses Chrome updates a bunch. Sorry for the trouble.

**Comment by denelon** [comment]
We've been discussing the behavior for the hash override behavior. "--force" doesn't really convey that you are taking a risk in installing a package when the hash doesn't match. I think we're leaning towards an administrative setting to enable the hash override. Then we will need to enable this mechanism to work both for user and admin modes as some programs don't work when installing under one or the other mode. We will also need to put out a very clear warning about the fact that the hash was ignored.

**Comment by Zamiell** [comment]
> Any other way to work around it would probably be a bug in UAC.

Is this actually true? For reference, the analog in Linux would be running the following command as root:

```bash
runuser -l james -c "winget install --accept-source-agreements --silent --exact --id Google.Chrome --force"
```

If I performed this command in Linux, it would notably **not** prompt me for the password of james. And this behavior does not violate the security model, because I already am the root user and therefore should have full access to the system. (Consider that if typing `passwd james` as the root user prompted me for the existing password of james before letting me assign a new password, that would be absurd.)

The specific issue here is that the `runuser` command is not a perfect analog to the `runas` command. Specifically, the latter forgets the context of which user is invoking it, where the former is a bit smarter.

All I mean to say by this is that a workaround for the Google Chrome issue seems like it could exist, in principle.

**Comment by jedieaston** [comment]
It could exist, you're right. But an easier place to fix it would be in winget. Windows only lets you [impersonate](https://docs.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-impersonateloggedonuser) a user if they are currently logged in, and if you know their password (PsExec would probably also work to do what you want, but you'd need the user's password to be entered just like runas).

The traditional *nix way of root knowing everything about everything makes life easier, which is why I still prefer using Linux for running servers and things. But I think the design assumption Microsoft made was the correct one ("why would someone need to impersonate a user without their permission? if they do have permission, the target user can definitely type their password in"), as under most cases people in the Administrators group can access all user's files/settings/etc, they just can't pretend to be that user while doing it.

**Comment by MarcoPeraza** [comment]
> The actual solution to your problem is that we need to be better about updating Chrome and Spotify (the two apps that change their installer hashes most often) in the community repo. I just saw the PR for it had been sitting for 24 hours or more before a moderator got it approved (oops), then it sat for another hour until the bot got it merged. I've gotten lazy due to the community automation being better than it used to be, but it still misses Chrome updates a bunch. Sorry for the trouble.

As long as there's any lag time at all during which the package is broken, then it's hard to use winget for any serious scripts. I don't see how an approach focused on shortening that time period, instead of eliminating it, can get you there.

**Comment by jedieaston** [comment]
I would argue that running "serious scripts" where there's no hash validation of binaries that may be running on important systems would make it hard to use winget though. I think if we can get to where brew is, where packages are updated within an hour or two of the ISV updating their server, we'll be in a healthy spot. (I'm pretty sure the reason brew can do that is a mixture of analytics from the client and good community participation, and we can get at least one of those easily.)

**Comment by MarcoPeraza** [comment]
I see the point regarding verification of binaries. But Microsoft has a very good code signing system baked into Windows that winget could leverage to provide those same guarantees. 

Maybe another way of putting it: what attack are you protecting against? Someone taking control of the software vendor's download URL and replacing it with malicious software? In that situation, users are screwed anyway. 1) auto-update mechanisms in the software are likely compromised too and 2) the user's likely recourse after winget failing is to go to the vendor's website... and download that same binary, and 3) they would also need to have compromised the code signing certificate.

**Comment by MarcoPeraza** [comment]
I set up a new machine today, and again, Google Chrome could not be installed due to a hash mismatch.

**Comment by flowoy96** [comment]
> I set up a new machine today, and again, Google Chrome could not be installed due to a hash mismatch.

Can confirm. Google Chrome won't work, even with "--force" flag when running as Administrator. Without admin it works fine with "--force" flag

**Comment by denelon** [comment]
Merged 6 hours ago:
* https://github.com/microsoft/winget-pkgs/pull/83280

**Comment by FaserF** [comment]
Same issue seems to be with Teamviewer and Teamviewer Host.

**Comment by Kmfernan5** [comment]
Same for Beeper and Miro. Has someone started a list?

**Comment by sschmocker** [comment]
Same Problem for us, chrome wont install because of mismatch that cant be overwritten as admin

**Comment by FaserF** [comment]
And I want to add that the same goes for Logi Options+

**Comment by cgerke** [comment]
> I don't think you need to have that concern. There are any number of things that if you copy and paste into a administrator command prompt will absolutely trash/compromise/etc. your computer. An admin command prompt can do _anything_ on a Windows machine (even perform an escalation to SYSTEM). So I don't see a point in being concerned about it in this one small case. Especially if someone has turned off UAC.

I agree with this position. Its not up to winget to decide on the security of the operating system. That is up to the overall security configuration deployed within your enterprise. There are multiple layers here already including but not limited to ASR. App Control and coming soon PIM solution for adminstrator elevation.

**Comment by cgerke** [comment]
> I see the point regarding verification of binaries. But Microsoft has a very good code signing system baked into Windows that winget could leverage to provide those same guarantees.
> 
> Maybe another way of putting it: what attack are you protecting against? Someone taking control of the software vendor's download URL and replacing it with malicious software? In that situation, users are screwed anyway. 1) auto-update mechanisms in the software are likely compromised too and 2) the user's likely recourse after winget failing is to go to the vendor's website... and download that same binary, and 3) they would also need to have compromised the code signing certificate. Like Raymond Chen says, "it rather involved being on the other side of this airtight hatchway".

This got me thinking too, we use Defender and Sentinel in our enterprise. Microsofts AI driver security system is extremely powerful, I would have thought if winget was becoming a core component of Intune that its Microsoft's intension to ensure the backend is protected by these same systems MS offers to enterprise and you'd have some kind of mechanism where Defender scans the manifests, downloads the executables and compares the hash against its Threat Protection system?

**Comment by denelon** [comment]
WinGet operates in several differnt kinds of environments. This issue is really about how we initially implemented the hash override mechanism. We agree it needs to change, but it's considered a breaking change which would require a "major" version bump.

This is actually the current top reason we would consider that activity. We're compiling a list of other potential reasons to justify a major version bump. It's not something we take lightly, and we also want to be very explicit about other changes being considered:
* https://github.com/microsoft/winget-cli/issues/2128

**Comment by FaserF** [comment]
@Skeb1ns I am Using the great tool "[WingetAutoUpdate](https://github.com/Romanitho/Winget-AutoUpdate)" in my company, which lets you set a "[Blacklist](https://github.com/Romanitho/Winget-AutoUpdate/blob/main/excluded_apps.txt)" for auto updates. For me in the blacklist are Chrome, Teamviewer and Office apps. 
Maybe that helps you.

To deploy winget apps I am using [winget-install ](https://github.com/Romanitho/Winget-Install)which works fine for me with Intune

**Comment by Trenly** [comment]
[Policy] Breaking-Change

**Comment by Trenly** [comment]
There is the `--ignore-security-hash` argument, but as the prompt states, it will not work when running as admin. When running in a non-elevated prompt though, it should work. This is an artifact from the original security requirements for winget if I recall correctly.

**Comment by eabase** [comment]
@denelon 

> We now have a Group Policy to determine if the ability to override hash validation is available to users. 

Where can we locate this Policy?

**Comment by baymax55** [comment]
same question for me
![image](https://github.com/microsoft/winget-cli/assets/35788491/7c5c3063-e901-4cb5-bee7-abce8ac2a667)

**Comment by eabase** [comment]
> @denelon
> 
> > We now have a Group Policy to determine if the ability to override hash validation is available to users.
> 
> Where can we locate this Policy?

Bump!

**Comment by Geczy** [comment]
`winget install -e --id Logitech.GHUB --source winget --force` doesn't work

```sh
C:\Windows\System32>winget install -e --id Logitech.GHUB --source winget --force
Found Logitech G HUB [Logitech.GHUB] Version 2023.6.430723
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://download01.logi.com/web/ftp/pub/techsupport/gaming/lghub_installer.exe
  ██████████████████████████████  39.6 MB / 39.6 MB
Installer hash does not match.
```

**Comment by AndreasBrostrom** [comment]
Tried this short command to attempt upgrade but it aint working:
`winget install --accept-source-agreements --exact --id Google.Chrome --force --ignore-security-hash`

```pwsh
Administrator in ~
PS > winget install --accept-source-agreements --exact --id Google.Chrome --force --ignore-security-hash
Found Google Chrome [Google.Chrome] Version 117.0.5938.63
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://dl.google.com/dl/chrome/install/googlechromestandaloneenterprise64.msi
  ██████████████████████████████   103 MB /  103 MB
Installer hash does not match; this cannot be overridden when running as admin
```

**Comment by z-impi** [comment]
same here. :(
beat my head against the wall. Found no solution yet...

**Comment by eabase** [comment]
following.. 👀

**Comment by uffemcev** [comment]
An attempt to automate this without interactive popups. These commands must be run as an administrator. Winget v1.6.2771.

```powershell
$id = "Google.Chrome"
winget settings --enable InstallerHashOverride
reg add "HKCU\Software\Microsoft\Windows\CurrentVersion\Policies\Associations" /v "LowRiskFileTypes" /t REG_SZ /d ".exe;.msi;.zip;" /f
runas /trustlevel:0x20000 /machine:amd64 "winget install --id=$id --accept-package-agreements --accept-source-agreements --ignore-security-hash --exact --silent"
```

**Comment by elisimpson** [comment]
I've found the issue is usually the manifest pointing to a "latest" or unversioned download link. These change frequently and of course the hash won't match with every change. Maybe we should disallow pointing to unversioned installers in the manifest, or have a special flag that allows "latest" installs without a hash.

**Comment by dennish78** [comment]
Of course the real solution is to force the hash to be part of the manifest. Why would a hash be non-mandatory? It's trivial to add.

**Comment by n-ice-ch** [comment]
> Of course the real solution is to force the hash to be part of the manifest. Why would a hash be non-mandatory? It's trivial to add.

That is the typical answer of a theorist.
Who should customize the hash for WinGet? Google?

If an administrator (practitioner), trust Google, then a download link to the latest available version is enough for. How should the administrator adjust the hash in the WinGet if the binary is changed (by Google!)?

The --force parameter must allow an administrator to force an installation. What else is the purpose of --force?

**Comment by dennish78** [comment]
> > Of course the real solution is to force the hash to be part of the manifest. Why would a hash be non-mandatory? It's trivial to add.
> 
> That is the typical answer of a theorist. Who should customize the hash for WinGet? Google?
> 
> If an administrator (practitioner), trust Google, then a download link to the latest available version is enough for. How should the administrator adjust the hash in the WinGet if the binary is changed (by Google!)?
> 
> The --force parameter must allow an administrator to force an installation. What else is the purpose of --force?

Not sure what you mean. The publisher of the app in the Winget repo is responsible for delivering both the software and the manifest, I'm not referring to the hash of Winget itself.
It's the responsibility of the software publisher to provide a correct and up to date manifest along with the actual installer.
If the installer cannot be verified, it has no business being on a software repository that is used by IT administrators.
We might as well go back to manually taking care of installers and ignoring Winget entirely.

I'd fire an admin on the spot if I saw them pushing 'winget upgrade --all --force' on thousands of computers if --force would bypass verification checks.

**Comment by Masamune3210** [comment]
A grand majority of the programs in winget's catalog were not and are not moderated by the developer of the program itself.....

**Comment by dennish78** [comment]
In that case it's an even better thing that --force doesn't let you skip the hash validation. If Winget is supposed to be the end all be all package management solution, validation is key. If not, it will be stuck in hobbyist land forever.
My point stands, whoever maintains the manifest needs to supply a valid hash. It's trivial to do (curl package, get-filehash).

**Comment by ninbura** [comment]
This become an annoying problem when utilizing ssh on Windows. When you ssh into another Windows machine you're connecting as an admin. In this situation it is require that you install an application via cli, and if that application's installer hash doesn't match you're simply out of luck.

I'm not sure I understand why there isn't just an additional flag you can provide. If the team feels that `--force` isn't descriptive enough; just create a new flag...

**Comment by dennish78** [comment]
Ooooh I love that chocolatey feature of being able to  provide a checksum.

**Comment by MarcoPeraza** [comment]
> > Not sure what you mean. The publisher of the app in the Winget repo is responsible for delivering both the software and the manifest, I'm not referring to the hash of Winget itself. It's the responsibility of the software publisher to provide a correct and up to date manifest along with the actual installer. If the installer cannot be verified, it has no business being on a software repository that is used by IT administrators. We might as well go back to manually taking care of installers and ignoring Winget entirely.
> > I'd fire an admin on the spot if I saw them pushing 'winget upgrade --all --force' on thousands of computers if --force would bypass verification checks.
> 
> I don't agree, and I think you are using an extreme case as an example that we are not talking about. No one with a healthy brain would do `widget upgrade --all --force,` and every example in this thread was about one concrete app installation/upgrade.
> 
> The override is useful when some concrete app upgrade fails (during `winget upgrade --all` for example) because the package contains an invalid hash.
> 
> In the world of all packages being updated, this would not be necessary, and I would agree that the package author should provide a proper hash. But the reality of the winget package repository is that many packages are outdated. And because many apps do not use versioned installer file names (i.e. URL stays the same when a new app version is published), the hash gets outdated because the executable stated in the manifest was updated in the remote location by the app developer while the corresponding package was not. Many packages are "maintained" by volunteers, not the app developers directly.
> 
> You might NEED to install the new app version, maybe because the new app version contains important security updates, important bug fixes, or whatever.
> 
> In that case, you want to tell Winget, "Hey, I know what I'm doing, and I'm ok with the risks involved," because the risk associated with using the old app version may be bigger for you. In that case, winget trying to be "secure" is not actually secure.
> 
> In that case, you are screwed. And what will you do in such a situation? You will probably go to the app homepage, download the installer ANYWAY, and install it manually if you evaluate the risks involved. So the result is the same.
> 
> `chocolatey` has a simple solution to this - it lets you provide your own hash via `--checksum64` or simply ignore the hash via `--ignore-checksums` - simple, effective, and useful. I think it's actually more secure because by using `--checksum64`, you can easily use a valid hash provided by the app homepage, so the installation of the new version is secure. When Winget does not provide that option, the user might ignore the checksum validation completely and install it right away, fingers crossed.
> 
> I think this is the classic example of an application trying to protect users from themselves, which, in the case of an application for administrators, does not make sense and is only annoying and can backfire when it does not provide a workaround.

100% agree. This behavior remains baffling and inexplicable. It’s been three years since I reported this issue that makes winget totally unreliable.  And there’s no solution in sight.

**Comment by pandadolphin** [comment]
Solved on Win11 by two steps on Powershell:
`winget settings --enable InstallerHashOverride` as admin
`winget install NirCmd --ignore-security-hash` as normal user

**Comment by uffemcev** [comment]
A new attempt to automation without interactive popups. These commands must be run in powershell as an administrator:
```
$id = "Google.Chrome"
winget settings --enable InstallerHashOverride
$env:SEE_MASK_NOZONECHECKS = 1
runas /trustlevel:0x20000 /machine:amd64 "winget install --id=$id --accept-package-agreements --accept-source-agreements --ignore-security-hash --exact --silent"
```
If anyone can test it with packages that don't have a matching hash I would appreciate it.

**Comment by eabase** [comment]
Not sure what's up ⬆️ with all those thumbs ⬇️.  
Isn't it obvious to everyone that we should be in charge of our own computers? 

There are many examples of why one should be able to install non conformant or hashed packages:
- A lot of security researchers may need to install malware on purpose.
- People porting SW from old machines and OS's with outdated certs. 
- ALL of what @dennish78  email quote said.
- I am testing something during development, and just can't be bothered with fixing hashes for installing my module for something that may be later part of another larger project.
- `NOBODY` should have to resort to Windows OS hacking to be able to run whatever we want on it. Having to edit registry items, and circumvent problematic security mechanisms, should be understood as a much greater threat, than preventing the owner of a machine the liberty to install whatever they like.  
*And let's be honest here, any security mechanism marketed to prevent user from getting attacked, are and will be easily overcome by black-hat experts.*


It seem to me it's more about the MS corporate idea that they have to take control of everything, which is why essential packages for outdated windows versions are being scrubbed from internet. (I.e. Windows-7,8 framework packages and updates.) Causing substantial security issues for users trying to recover, port and update old software, as they have to find very sketchy sources for their MS updates packages. To do that you need to first understand how it works on the old machines, but if MS is preventing everyone... (Is anyone in MS thinking there!?)

**Comment by Masamune3210** [comment]
1. Malware wouldn't be allowed in winget in the first place
2. Winget wouldn't be an option there anyway
3. Hashes are already part of the standard that all manifests adhere to
4. You wouldn't be using Winget to install something that is already local, and especially not something you are actively iterating on. 
5. This is the only part I agree with, although I'm not sure where you got the idea that you have to change things in the registry to get anything involving Winget or things it downloads to run. Also, no, that's not how that works, there are plenty of security mechanisms that have been proven so far as to be effective in what they intend. 

Sidenote - Nobody is scrubbing old files from the internet, websites are just the definition of lazy in a lot of cases and hotlink to files instead of doing what they are supposed to and hosting it locally, which means as soon as the origin link is dead, anything that hotlinked to it is dead as well. As to the whole "sketchy sources" part, unless you are looking at files for exceedingly old versions of Windows, all files are signed and checked before install.

**Comment by eabase** [comment]
1. You're missing the point... (Go ask your local White Hat.)
2. Sometimes we need to run OLD SW on new machines.
3. You're not really saying anything there. 🤷‍♂  
4. Sounds more like you're trying to tell me how to do my development, rather than support what people actually do... 
5. That's what you have to do to run on no longer supported machines / Win OS's.  
   (Clearly You never tried, or you would not have said this.)

> *Nobody is scrubbing old files from the internet,* 

LoL! I have an entire Excel full of scrubbed MS links and where to find what was there before. And I'm certainly not gonna share that here! But I can give you cookie crumb.

Try to download this: 
* https://www.microsoft.com/en-us/download/details.aspx?id=54616
* ﻿https://www.microsoft.com/en-us/download/confirmation.aspx?id=40855﻿
* ⬆️ also mangled from WebArchive [here](https://web.archive.org/web/20181126124429/https://www.microsoft.com/en-us/download/confirmation.aspx?id=40855)

Case Example:  
You'd be extremely surprised by the performance of `Windows-7` when running powershell, and finding that your powershell script that normally takes **20 seconds** to run in `Windows-11`, only takes **3 seconds** in `Windows-7`!

**Comment by Masamune3210** [comment]
I'm genuinely not sure if you are just misunderstanding what Winget is or whats going on here, but 

A. The oldest version of Windows that it supports is Windows 10 Build 16299
B. It has never accepted malware and will never accept malware. 
C. Has nothing to do with dev because its for downloading already built, fully complete installers from other places on the internet.
D. Doesn't require registry muckery because it can't due to the tests that manifests have to pass before they are even accepted into the repo. All manifests are tested to function correctly in a completely unaltered Windows VM before they even make it into the public manifests.

**Comment by trivail-mike-davis** [comment]
> A new attempt to automation without interactive popups. These commands must be run in powershell as an administrator:
> 
> ```
> $id = "Google.Chrome"
> winget settings --enable InstallerHashOverride
> $env:SEE_MASK_NOZONECHECKS = 1
> runas /trustlevel:0x20000 /machine:amd64 "winget install --id=$id --accept-package-agreements --accept-source-agreements --ignore-security-hash --exact --silent"
> ```
> 
> If anyone can test it with packages that don't have a matching hash I would appreciate it.

That didn't work for me, but it appears they're now recommending the following method instead which does work for me:

https://blog.intunepckgr.com/2024/06/19/using-the-google-chrome-exe-deployment-from-winget-over-the-standard-google-chrome-msi/

```
.\winget.exe install --id Google.Chrome.EXE --exact --scope machine --architecture x64 --silent --disable-interactivity --accept-package-agreements --accept-source-agreements
```

**Comment by uffemcev** [comment]
> That didn't work for me
Hi! Can you describe what exactly didn't work for you please? Maybe some logs? 🤔

**Comment by crazivik** [comment]
New method as per [trivail-mike-davis](https://github.com/trivail-mike-davis) works!!

https://blog.intunepckgr.com/2024/06/19/using-the-google-chrome-exe-deployment-from-winget-over-the-standard-google-chrome-msi/

**Comment by Dontoronto** [comment]
> Solved on Win11 by two steps on Powershell: `winget settings --enable InstallerHashOverride` as admin `winget install NirCmd --ignore-security-hash` as normal user

@pandadolphin you are the chosen one sent to bring an end to the suffering

**Comment by intruder777** [comment]
This is completely unacceptable situation when you can't override that security settings while being on elevated shell.
It causes the following "**dead loop**" situation:
If you're installing something to machine scope you need to be elevated (in order to have write access to Program Files). But if the package you're trying to install has hash issues and you have to use **--ignore-security-hash** winget tells you you must be non-elevated. Dead end! So you end up not being able to install that package at all!
Even if you decide to move your **portablePackageMachineRoot** out of **Program Files** dir to some location where non-elevated user has access - it sill fails, because winget still trying to create a symlink in **%ProgramFiles%\WinGet\Links** folder and naturally fails again. And as you may know, winget doesn't allow you to configure **Links** custom directory location.
That's a total dead end!

P.S.: When user intentionally chooses to ignore hash mismatch - it's completely their responsibility. They can still go manually  download that invalid hash installer, run it elevated, and still get themselves in trouble. So I don't understand why such stupid winget restriction even exists.

P.P.S.: Also, unelevated winget can't access HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\ registry key. So currently there is no way to normally install package with a hash mismatch to machine scope.

**Comment by n-ice-ch** [comment]
It's unbelievable how blatantly Microsoft is trying to prevent “side loading” outside of its Store!

It's obvious what's going on here. This has absolutely nothing to do with security. It's pure harassment of administrators—and, not least, of users.

**Comment by denelon** [comment]
Thank you all for the sustained, detailed discussion here — the feedback has been genuinely useful in shaping how we think about this. Sharing where we've landed and the direction we're leaning.

**What happens today**

The hash-mismatch override already exists, but it's gated:
- Enable the admin setting (`winget settings --enable InstallerHashOverride`), which is also controllable via Group Policy for IT, then pass `--ignore-security-hash` on a **non-elevated** command.
- When running **elevated/as admin**, the override is hard-blocked — no flag or setting bypasses it. That's the behavior most of this thread is running into.

**How we're thinking about it**

The original "no override when elevated" rule was about blast radius: a mismatched binary run with admin rights can do the most damage, so we disabled the override where the risk is highest. We still take installer integrity seriously and want safe-by-default behavior. That said, we recognize a few real problems with it as an absolute rule:

- **It blocks a legitimate, supported path.** Machine-scope installs require elevation, so a hash mismatch there currently can't be overridden at all — even deliberately, one package at a time. That's a dead-end we don't want.
- **It pushes people toward less-safe workarounds** (`runas` tricks, `SEE_MASK_NOZONECHECKS`, registry edits, third-party wrappers) that are harder to audit than a first-class, warned, policy-gated override.
- **Most of the pain isn't really about security** — it's manifest staleness, usually from unversioned/"latest" installer URLs whose hash drifts when the publisher updates the binary before the manifest catches up. We're continuing to reduce that lag on the winget-pkgs side, but it will never be zero, so a deliberate escape hatch is warranted.

**Direction we're leaning**

- Keep hash validation on by default; `--force` will **not** silently bypass integrity (especially for bulk operations like `upgrade --all`).
- Allow a **deliberate, single-package** override via `--ignore-security-hash` that works **whether or not** you're elevated, still gated behind the `InstallerHashOverride` admin setting **and** Group Policy (so enterprises can hard-disable it), with a clear warning and telemetry.
- Explore a **supply-your-own-hash** option (similar to Chocolatey's `--checksum`) so a newer version can be verified against a vendor-published hash rather than skipping verification entirely — preserving integrity for the "the package is just stale" case.

**Sequencing**

Changing the elevated hard-block is a behavioral (breaking) change, so we're weighing it carefully — but we're open to bringing it in rather than indefinitely deferring it. It's part of the broader breaking-changes discussion in #2128.

Keeping this open to track that work.

*Drafted with the help of GitHub Copilot.*
